include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -778,6 +778,33 @@ class FEEvaluationBase
    * MatrixFree object was given at initialization.
    */
   mutable std::vector<types::global_dof_index> local_dof_indices;
+
+  /**
+   * A temporary data structure for the Jacobian information necessary if
+   * the reinit() function is used that allows to construct user batches.
+   */
+  AlignedVector<Tensor<2, dim, VectorizedArrayType>> jacobian_data;
+
+  /**
+   * A temporary data structure for the Jacobian determinant necessary if
+   * the reinit() function is used that allows to construct user batches.
+   */
+  AlignedVector<VectorizedArrayType> J_value_data;
+
```

Comment:
Thinking about these and the other two members below, I'm wondering whether it wouldn't make to expand https://github.com/dealii/dealii/blob/5df4e622f788a831734f48871e25f22115b10b23/include/deal.II/matrix_free/fe_evaluation_data.h#L865-L867 instead, as that one has all necessary members already and would therefore avoid bloating the interface here. Of course, we would need to add an additional constructor there and avoid building the `FEValues` object in that field, but overall I would be much happier of not having to look out on 3 different locations for this.

Reply:
That sounds like a good idea! I would suggest that we proceed with this PR as it is and do the suggestion in a follow-up PR - it does not influence the behavior to the outside! I need to take  a more detailed look at `internal::MatrixFreeFunctions::MappingDataOnTheFly`. Once we have converged on a solution, I would like to add a similar function to `FEFaceEvaluation`!

Reply:
Indeed, we need to check how to solve the case for `FEFaceEvaluation` as well. I believe things can be done with rather small adaptations (constructors, possibly the way we grant other classes access to `MappingDataOnTheFly`, etc). Another thing we need to consider is how to possibly re-use the same mapping data between different `FEEvaluation` objects, relating also to the reuse discussion we are currently having in #13245.